provisioning/service/src/Contract/ContractApiHttp.cs

```diff
@@ -487,143 +165,44 @@ private static void ValidateHttpResponse(ContractApiResponse response)
-
-                    throw new ProvisioningServiceClientException(ex.Message, ex);
-                }
-                catch (TimeoutException ex)
-                {
-                    throw new ProvisioningServiceClientTransportException(ex.Message, ex);
-                }
-                catch (IOException ex)
-                {
-                    throw new ProvisioningServiceClientTransportException(ex.Message, ex);
-                }
-                catch (HttpRequestException ex)
-                {
-                    throw new ProvisioningServiceClientTransportException(ex.Message, ex);
-                }
-                catch (TaskCanceledException ex)
-                {
-                    // Unfortunately TaskCanceledException is thrown when HttpClient times out.
-                    if (cancellationToken.IsCancellationRequested)
-                    {
-                        throw new ProvisioningServiceClientException(ex.Message, ex);
-                    }
-
-                    throw new ProvisioningServiceClientTransportException(string.Format(CultureInfo.InvariantCulture, "The {0} operation timed out.", httpMethod), ex);
-                }
-                catch (Exception ex)
-                {
-                    //if (Fx.IsFatal(ex)) throw;
-
-                    throw new ProvisioningServiceClientException(ex.Message, ex);
-                }
+                return;
+            }
 
-                if (isMappedToException(responseMsg))
-                {
-                    Exception mappedEx = await MapToExceptionAsync(responseMsg, mergedErrorMapping).ConfigureAwait(false);
-                    throw mappedEx;
-                }
+            if (!ifMatch.StartsWith("\"", StringComparison.OrdinalIgnoreCase))
+            {
+                ifMatch = "\"" + ifMatch;
             }
-        }
 
-        private static async Task<Exception> MapToExceptionAsync(
-            HttpResponseMessage response,
-            IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> errorMapping)
-        {
-            Func<HttpResponseMessage, Task<Exception>> func;
-            if (!errorMapping.TryGetValue(response.StatusCode, out func))
+            if (!ifMatch.EndsWith("\"", StringComparison.OrdinalIgnoreCase))
             {
-                return new ProvisioningServiceClientException(
-                    await ExceptionHandlingHelper.GetExceptionMessageAsync(response).ConfigureAwait(false),
-                    isTransient: true);
+                ifMatch = ifMatch + "\"";
             }
 
-            var mapToExceptionFunc = errorMapping[response.StatusCode];
-            var exception = mapToExceptionFunc(response);
-            return await exception.ConfigureAwait(false);
+            requestMessage.Headers.IfMatch.Add(new EntityTagHeaderValue(ifMatch));
         }
 
         /// <summary>
         /// Release all HTTP resources.
         /// </summary>
         public void Dispose()
```

Comment:
This will fail FXCop. 
See https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern for the correct pattern with the `protected void Dispose(bool)` function.

Reply:
done